### ISSUE TITLE
"No such image: testcontainers/ryuk:0.3.0"

### ISSUE THREAD

**Issue by gesellix** [testcontainers/testcontainers-java#3574]
Testcontainers 1.15.0 on Docker Engine 20.10/Docker for Mac 2.5.4 fails with the following stacktrace:

```
org.testcontainers.containers.ContainerLaunchException: Container startup failed

	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:327)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:308)
	at org.testcontainers.spock.TestcontainersMethodInterceptor.startContainers_closure3(TestcontainersMethodInterceptor.groovy:83)
	at groovy.lang.Closure.call(Closure.java:405)
	at groovy.lang.Closure.call(Closure.java:421)
	at org.testcontainers.spock.TestcontainersMethodInterceptor.startContainers(TestcontainersMethodInterceptor.groovy:80)
	at org.testcontainers.spock.TestcontainersMethodInterceptor.interceptSetupSpecMethod(TestcontainersMethodInterceptor.groovy:25)
	at org.spockframework.runtime.extension.AbstractMethodInterceptor.intercept(AbstractMethodInterceptor.java:36)
	at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:97)
	at org.spockframework.spring.SpringInterceptor.interceptSetupSpecMethod(SpringInterceptor.java:37)
	at org.spockframework.runtime.extension.AbstractMethodInterceptor.intercept(AbstractMethodInterceptor.java:36)
	at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:97)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)
Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=couchdb:1.7.2, imagePullPolicy=DefaultPullPolicy())
	at org.testcontainers.containers.GenericContainer.getDockerImageName(GenericContainer.java:1278)
	at org.testcontainers.containers.GenericContainer.logger(GenericContainer.java:612)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:317)
	... 16 more
Caused by: com.github.dockerjava.api.exception.NotFoundException: Status 404: {"message":"No such image: testcontainers/ryuk:0.3.0"}

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:241)
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:125)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:33)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:13)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.CreateContainerCmdImpl.exec(CreateContainerCmdImpl.java:595)
	at org.testcontainers.utility.ResourceReaper.start(ResourceReaper.java:91)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:203)
	at org.testcontainers.LazyDockerClient.getDockerClient(LazyDockerClient.java:14)
	at org.testcontainers.LazyDockerClient.listImagesCmd(LazyDockerClient.java:12)
	at org.testcontainers.images.LocalImagesCache.maybeInitCache(LocalImagesCache.java:68)
	at org.testcontainers.images.LocalImagesCache.get(LocalImagesCache.java:32)
	at org.testcontainers.images.AbstractImagePullPolicy.shouldPull(AbstractImagePullPolicy.java:18)
	at org.testcontainers.images.RemoteDockerImage.resolve(RemoteDockerImage.java:66)
	at org.testcontainers.images.RemoteDockerImage.resolve(RemoteDockerImage.java:27)
	at org.testcontainers.utility.LazyFuture.getResolvedValue(LazyFuture.java:17)
	at org.testcontainers.utility.LazyFuture.get(LazyFuture.java:39)
	at org.testcontainers.containers.GenericContainer.getDockerImageName(GenericContainer.java:1276)
	... 18 more
```

**Comment by bsideup** [comment]
@gesellix do you have a mirror configured? Or is this a response from Docker Hub?

**Comment by gesellix** [comment]
I suppose that the 404 is the local Docker Engine and no lookup to any registry is made.

**Comment by gesellix** [comment]
As far as I know the docker cli had some logic like this for `docker run` (pseudo code):

```
try {
  create _container(image)
}catch(e){
  if e.status == 404 {
    pull_image(image)
    create _container(image)
  }
}
```

I don't understand, though, why we now run into such an issue. The only thing I'm aware of is https://github.com/docker/cli/pull/1498, which _might_ be related.

**Comment by gesellix** [comment]
Did/does Testcontainers pull images, in this case ryuk, before trying to create containers?

**Comment by keeganwitt** [comment]
I'm hitting this today too.  In my case, for a Postgres container.  I tried setting `ryuk.testcontainer.image=testcontainersofficial/ryuk:0.3.0`. It couldn't pull that image either. They definitely exist in Docker Hub though.

The other thing I suspected was maybe we're hitting the Docker pull limits?  I thought since this is communicating with the daemon it should use the auth configured on the host, but possibly I'm misunderstanding.

**Comment by gesellix** [comment]
... so maybe the pull for ryuk is performed unauthenticated?

**Comment by keeganwitt** [comment]
That's what I was worried about, yea.  But I'm not sure if that's what's going on or not yet.

**Comment by bsideup** [comment]
@gesellix we definitely pull the image if it is not available. 

> The other thing I suspected was maybe we're hitting the Docker Hub pull limits?

In that case, an error would differ (unless the Docker Hub team have decided that `404` is a perfect http status code for a rate limited response, which should be 5xx instead)

**Comment by keeganwitt** [comment]
When we got the error before, from running docker commands directly in a job, we got a message specifically that said we hit pull limits.  But I don't know the HTTP status that the docker binary received in that case, so I wasn't sure if the message was possibly being hidden by TestContainers or not.

**Comment by gesellix** [comment]
While our GitHub Actions still work (same Testcontainers version, but differend Docker Engine/operating system), I guess this is mainly related to Docker for Mac. I can give it a try with an older Docker4Mac release tomorrow.

**Comment by keeganwitt** [comment]
In my case, it's passing locally on Mac (but I have those images in my cache though) and failing on GitLab.

**Comment by bsideup** [comment]
404 and `{"message":"No such image: testcontainers/ryuk:0.3.0"}` is what we actually get from the API.

Also, `testcontainers/*` images are exempt from rate limiting, or at least that's what they told us :)

**Comment by bsideup** [comment]
oh wait, I think I know what is it...

**Comment by keeganwitt** [comment]
Maybe Docker Hub is having some problem?  I just tried disabling ryuk and then it said `no such image: alpine:3.5`

**Comment by gesellix** [comment]
For me Docker Hub seemed to be ok, `docker pull *ryuk` made it work for me... well... maybe the oder images have been in the local cache 🤔

**Comment by keeganwitt** [comment]
Good point. `docker pull` didn't break for me either locally.

**Comment by gesellix** [comment]
https://status.docker.com/pages/533c6539221ae15e3f000031 looks ok

**Comment by bsideup** [comment]
Ok, "filter by image name" query parameter in `/images/json` got removed, and now this condition fails:
https://github.com/testcontainers/testcontainers-java/blob/d135a2605401f6c663aab4e7edc6d6d76716f930/core/src/main/java/org/testcontainers/DockerClientFactory.java#L330

I just submitted #3575 with a fix, will be included in 1.15.1

**Comment by keeganwitt** [comment]
I suppose I should mention too that I had tried with 1.15.0-rc2 and 1.15.0.

**Comment by keeganwitt** [comment]
> 
> 
> Ok, "filter by image name" query parameter in `/images/json` got removed, and now this condition fails:
> 
> https://github.com/testcontainers/testcontainers-java/blob/d135a2605401f6c663aab4e7edc6d6d76716f930/core/src/main/java/org/testcontainers/DockerClientFactory.java#L330
> 
> I just submitted #3575 with a fix, will be included in 1.15.1

Ah, so a change in Docker Hub API?

**Comment by bsideup** [comment]
@keeganwitt Docker's API. Although the query param was deprecated (I wish we could run Docker in a strict API mode - will explore)

Sorry for this. We will release a hotfix ASAP. Meanwhile, consider pre-pulling `testcontainers/ryuk:0.3.0` and `alpine:3.5` :(

**Comment by keeganwitt** [comment]
> 
> 
> @keeganwitt Docker's API. Although the query param was deprecated (I wish we could run Docker in a strict API mode - will explore)
> 
> Sorry for this. We will release a hotfix ASAP. Meanwhile, consider pre-pulling `testcontainers/ryuk:0.3.0` and `alpine:3.5` :(

Yea, I'd thought of that, but I'm not sure it's possible with GitLab's Docker Executor.  It should be possible to run it as a script with shell executor instead though I suppose.  I'm still confused why it worked locally after deleting the ryuk image though...

**Comment by gesellix** [comment]
Thanks @bsideup for the quick fix!

**Comment by DaspawnW** [comment]
Are you planning to backport this to work also with junit 4?

**Comment by bsideup** [comment]
@DaspawnW this is not junit specific and, once released, will work with any type of integration (junit4, junit jupiter, spock, manual container lifecycle)

**Comment by arhohuttunen** [comment]
@keeganwitt did you ever find a reasonable workaround for builds running in Gitlab? We have been looking at this for a day now without much success. It works if you manually pre-pull the images, but we are using docker-machine to autoscale the runners in EC2, so manual work is not really an option.

**Comment by jdelucaa** [comment]
@bsideup I am seeing this also in 1.14.0.

**Comment by bsideup** [comment]
@jdelucaa yes, this Docker API change applies to most of Testcontainers versions.

**Comment by keeganwitt** [comment]
> 
> 
> @keeganwitt did you ever find a reasonable workaround for builds running in Gitlab? We have been looking at this for a day now without much success. It works if you manually pre-pull the images, but we are using docker-machine to autoscale the runners in EC2, so manual work is not really an option.

Not really.  I have exactly the same setup.  For now, we just commented the tests out, since a fix is forthcoming.  A few ideas came to mind, but I haven't really thought through them yet.

1. Customize the AMI to bake in the images.  Not totally sure if that'd work, and seems like a bad idea.
1. Put the docker pull commands as a user data script.
1. Don't use the Docker Executor, and instead use the Shell Executor and first pull the images.  Assuming that works, you'd have to specify the executor on the runner itself, so you'd need a separate set of runners to do this (which is possible, just give those runners a different tag).
1. Use DinD (Docker in Docker), and execute your build in a Dockerfile, rather than a script that runs on an image.  Might take some fiddling to get the volumes mounted correctly, I've never tried to do that before.

None of these seemed great.  If one of them sounds promising, I can explain in a little more detail what I was thinking, though there may be gotchas I haven't thought of.

**Comment by arhohuttunen** [comment]
We thought about option 1, but quickly discarded that idea. We also tried option 2 but apparently Docker isn’t installed at that point yet, so didn’t really proceed further with that. Neither 3 or 4 felt like a good idea, so I guess we’ll just skip the tests using testcontainers for now.

Thanks for sharing, though.

**Comment by keeganwitt** [comment]
@arhohuttunen I'm now thinking this broke because we upgraded GL Runners this week, which upgraded Docker version.  So downgrading should fix that. Unless others didn't upgrade and still ran into this?

I upgraded Docker for Mac to 3.0 (which has Docker 20.10 in it), and the test now fails locally too.

**Comment by arhohuttunen** [comment]
@keeganwitt I think on the runners we are using the docker stable tag, which should still point to 19.03.14 according to this:
https://hub.docker.com/layers/docker/library/docker/stable/images/sha256-8f71deccd0856d8a36db659a8c82894be97546b47c1817de27d5ee7eea860162?context=explore

**Comment by keeganwitt** [comment]
Sorry, I didn't mean the runners image, I meant the machines on which the runner image runs (where the daemon lives).  We use https://github.com/npalm/terraform-aws-gitlab-runner, which would upgrade that.
I've been talking with another of our engineers and he said userdata isn't the same as ec2-userdata. I didn't realize there were 2.

**Comment by Urokhtor** [comment]
We use a hard-coded AMI in our runners and haven't changed that lately, so that should not be the root cause in our case.

**Comment by keeganwitt** [comment]
Actually, some builds passed after our upgrade too, so it shouldn't have been that. I'm confused.

**Comment by Urokhtor** [comment]
I'll take previous comment back because those runners install Docker from official repo which does serve 20.10.

**Comment by jeantil** [comment]
I'm encountering this issue on apaceh CI builds for the apache/james project (https://builds.apache.org/blue/organizations/jenkins/james%2FApacheJames/detail/PR-268/16/pipeline) 
I tried pulling the image explicitly before running the tests but it still fails. we are very much looking forward to the 1.15.1 hotfix

**Comment by keeganwitt** [comment]
@jeantil That's correct, testcontainers is the default, testcontainersofficial was just a misguided thing I attempted early on (overriding the image in testcontainers properties). I saw testcontainersofficial mentioned in an issue where they were discussing Docker Hub and Quay.  Sorry for the confusion.

**Comment by mderouet** [comment]
When can we expect to have this 1.15.1 issue ? (In order to know whether we need to find a workaround for this or just wait for it)

**Comment by bsideup** [comment]
@mderouet the release is expected for later today (tsss ;))

**Comment by bsideup** [comment]
released in 1.15.1 🎉

**Comment by telapo** [comment]
It works, thanks

**Comment by alex-sky-cloud** [comment]
I got an update from Docker today, and I got **Docker desktop-3.0.1 (50773)**. 

Now I get an error while running Test container. 

`ru.sbrf.dfm.unit.ResourceServiceTest  Time elapsed: 2.309 s  <<< ERROR!
org.testcontainers.containers.ContainerLaunchException: Container startup failed
Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=postgres:11.7, imagePullPolicy=DefaultPullPolicy())
Caused by: com.github.dockerjava.api.exception.NotFoundException: 
{"message":"No such image: testcontainersofficial/ryuk:0.3.0"}`


I assumed you fixed it. I also disabled the **Use gRPC FUSE for file sharing** option, but it didn't helpHow can I fix this ?

**Comment by bsideup** [comment]
@alex-sky-cloud this is another issue, unrelated to the file sharing, and it is fixed in 1.15.1, please update.

**Comment by alex-sky-cloud** [comment]
I'm sorry. What should I update ?

**Comment by bsideup** [comment]
@alex-sky-cloud the project you're reporting to - Testcontainers :D

**Comment by alex-sky-cloud** [comment]
I have the latest version of docker.

Which project should I update ?
 
I executed the 

[docker system prune-af ](url)

command and only after that the error disappeared. 

So, is this how you will need to do it every time ?

**Comment by bsideup** [comment]
@alex-sky-cloud No. Just use the latest (1.15.1) version of Testcontainers.

**Comment by davoutuk** [comment]
This issue is still live for me when I tried using this on a Spring Boot/Postgres app.   The fix of separating downloading the 'testcontainer' docker image works.  
 I would suggest adding a note to the docs to explain that this is a pre requiste

**Comment by bsideup** [comment]
@davoutuk with Testcontainers 1.15.1? There was a bug that got fixed in 1.15.1, there isn't such prerequisite

**Comment by XXXAlexey** [comment]
just pull it manualy by console "docker pull testcontainers/ryuk:0.3.0"

**Comment by hafizali05** [comment]
i understand this might not work for every case but what fixed for me was to clean purge data and reset to factory settings

is this the best way? i dont know as i am new to docker and test containers.

![image](https://user-images.githubusercontent.com/10105388/103406469-b1f5a780-4b52-11eb-8ee8-13d541139a83.png)

**Comment by bsideup** [comment]
@hafizali05 this is not usually needed, unless your images cache is corrupted.

**Comment by hafizali05** [comment]
> @hafizali05 this is not usually needed, unless your images cache is corrupted.

i cant reproduce this bit again so if this persist again i might try to find out a way to only clear image cache. this ofcourse removed all my images and containers. @bsideup

**Comment by jeantil** [comment]
I confirm that after upgrading to test containers 1.15.1 we no longer have this issue on apache/james-project

**Comment by jeffnyauke** [comment]
I confirm that after upgrading to test containers 1.15.1 I no longer have this issue

**Comment by PauloPortfolio** [comment]
In my case, I have solved this problem cleaning all my docker-images.

However, upgrading, my POM.XML, to test containers 1.15.1, already, is a solution to his trouble.

**Comment by MiguelAngel82** [comment]
Hi. It seems that this error is happening again.
With the following versions:

- Docker Desktop for Mac 3.3.1. 
- Docker engine 20.10.5
- Testcontiners 1.15.3
- Gradle 5.4

It's failing with the same error:

```
java.lang.RuntimeException: Could not create new instance of class org.jboss.arquillian.test.impl.EventTestRunnerAdaptor
	at org.jboss.arquillian.test.spi.SecurityActions.newInstance(SecurityActions.java:146)
	at org.jboss.arquillian.test.spi.SecurityActions.newInstance(SecurityActions.java:89)
	at org.jboss.arquillian.test.spi.TestRunnerAdaptorBuilder.build(TestRunnerAdaptorBuilder.java:49)
	at org.jboss.arquillian.junit.AdaptorManager.initializeAdaptor(AdaptorManager.java:21)
	at org.jboss.arquillian.junit.ArquillianTestClassLifecycleManager.beforeTestClassPhase(ArquillianTestClassLifecycleManager.java:24)
	at org.jboss.arquillian.junit.ArquillianTestClass$1.evaluate(ArquillianTestClass.java:39)
	at org.spockframework.runtime.extension.builtin.ClassRuleInterceptor.intercept(ClassRuleInterceptor.java:39)
	at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:97)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:118)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:175)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:157)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at org.jboss.arquillian.test.spi.SecurityActions.newInstance(SecurityActions.java:144)
	... 28 more
Caused by: com.github.dockerjava.api.exception.NotFoundException: Status 404: {"message":"No such image: testcontainers/ryuk:0.3.0"}

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:241)
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:125)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:33)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:13)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.CreateContainerCmdImpl.exec(CreateContainerCmdImpl.java:595)
	at org.testcontainers.utility.ResourceReaper.start(ResourceReaper.java:91)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:203)
	at com.adaptavist.testfixtures.testcontainers.TestHostApplicationContainerProvider.<init>(TestHostApplicationContainerProvider.groovy:23)
	at com.adaptavist.testfixtures.application.BaseTestHostApplication.<init>(BaseTestHostApplication.groovy:18)
	at com.adaptavist.testfixtures.application.JiraCoreTestHostApplication.<init>(JiraCoreTestHostApplication.groovy:9)
	at com.adaptavist.arquillian.TestHostApplicationInstanceRegistrar.registerScriptRunnerFixture(TestHostApplicationInstanceRegistrar.groovy:20)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.ManagerImpl.start(ManagerImpl.java:253)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.<init>(EventTestRunnerAdaptor.java:61)
	... 29 more
```

I've tried to downgrade Docker Desktop and it was failing until Docker Desktop Community 2.5.0.1, which is using Docker engine 19.03.13. So with this last version (and keeping the same Testcontainers versions) is working.

Also, I've applied the workaround of manually downloading the image `testcontainers/ryuk:0.3.0` and with this workarodund is working for the last Docker Desktop version (3.3.1).

So please, if someone else is facing the same problem, maybe it's worth to have a look if something has broken with this last Docker version.

Thank you!

**Comment by bsideup** [comment]
@MiguelAngel82 Testcontainers 1.15.3 depens on Ryuk 0.3.1, meaning that you're still using an older version of Testcontainers

**Comment by MiguelAngel82** [comment]
@bsideup Thank you for your response! Sorry, it was my fault. Thanks to your response I've dug deeper into the project and I've realised that we were using another Testcontainers dependency, `org.testcontainers:selenium` that was not updated. 
So now, both dependencies has been updated to the latest version (1.15.3) and it's properly working ;)

Thank you and I apologise for the inconvenience.

**Comment by GigBucket** [comment]
> Testcontainers 1.15.0 on Docker Engine 20.10/Docker for Mac 2.5.4 fails with the following stacktrace:
> 
> ```
> org.testcontainers.containers.ContainerLaunchException: Container startup failed
> 
> 	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:327)
> 	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:308)
> 	at org.testcontainers.spock.TestcontainersMethodInterceptor.startContainers_closure3(TestcontainersMethodInterceptor.groovy:83)
> 	at groovy.lang.Closure.call(Closure.java:405)
> 	at groovy.lang.Closure.call(Closure.java:421)
> 	at org.testcontainers.spock.TestcontainersMethodInterceptor.startContainers(TestcontainersMethodInterceptor.groovy:80)
> 	at org.testcontainers.spock.TestcontainersMethodInterceptor.interceptSetupSpecMethod(TestcontainersMethodInterceptor.groovy:25)
> 	at org.spockframework.runtime.extension.AbstractMethodInterceptor.intercept(AbstractMethodInterceptor.java:36)
> 	at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:97)
> 	at org.spockframework.spring.SpringInterceptor.interceptSetupSpecMethod(SpringInterceptor.java:37)
> 	at org.spockframework.runtime.extension.AbstractMethodInterceptor.intercept(AbstractMethodInterceptor.java:36)
> 	at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:97)
> 	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
> 	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
> 	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
> 	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
> 	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)
> Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=couchdb:1.7.2, imagePullPolicy=DefaultPullPolicy())
> 	at org.testcontainers.containers.GenericContainer.getDockerImageName(GenericContainer.java:1278)
> 	at org.testcontainers.containers.GenericContainer.logger(GenericContainer.java:612)
> 	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:317)
> 	... 16 more
> Caused by: com.github.dockerjava.api.exception.NotFoundException: Status 404: {"message":"No such image: testcontainers/ryuk:0.3.0"}
> 
> 	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:241)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:125)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:33)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:13)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.command.CreateContainerCmdImpl.exec(CreateContainerCmdImpl.java:595)
> 	at org.testcontainers.utility.ResourceReaper.start(ResourceReaper.java:91)
> 	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:203)
> 	at org.testcontainers.LazyDockerClient.getDockerClient(LazyDockerClient.java:14)
> 	at org.testcontainers.LazyDockerClient.listImagesCmd(LazyDockerClient.java:12)
> 	at org.testcontainers.images.LocalImagesCache.maybeInitCache(LocalImagesCache.java:68)
> 	at org.testcontainers.images.LocalImagesCache.get(LocalImagesCache.java:32)
> 	at org.testcontainers.images.AbstractImagePullPolicy.shouldPull(AbstractImagePullPolicy.java:18)
> 	at org.testcontainers.images.RemoteDockerImage.resolve(RemoteDockerImage.java:66)
> 	at org.testcontainers.images.RemoteDockerImage.resolve(RemoteDockerImage.java:27)
> 	at org.testcontainers.utility.LazyFuture.getResolvedValue(LazyFuture.java:17)
> 	at org.testcontainers.utility.LazyFuture.get(LazyFuture.java:39)
> 	at org.testcontainers.containers.GenericContainer.getDockerImageName(GenericContainer.java:1276)
> 	... 18 more
> ```

I also has the erro, update the org.testcontainers jar version in maven pom.xml, then fixed!

**Comment by MelleD** [comment]
I'm experiencing the following error, which looks very similar.

The first time the Postgres container downloads and starts correctly. Sometimes (not always) the second time it starts, the error message appears. The error (often) occurs repeatedly on the build server, but it can't be reproduced 100%.


```
2025-05-14T06:22:06.1927466Z [INFO] Connected to docker: 
2025-05-14T06:22:06.1928448Z   Server Version: 28.0.4
2025-05-14T06:22:06.1928795Z   API Version: 1.48
2025-05-14T06:22:06.1929336Z   Operating System: Ubuntu 22.04.5 LTS
2025-05-14T06:22:06.1929706Z   Total Memory: 15989 MB
2025-05-14T06:22:06.2032564Z [INFO] Using ***/testcontainers/ryuk:0.11.0 as a substitute image for testcontainers/ryuk:0.11.0 (using image substitutor: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor'))
2025-05-14T06:22:06.2246667Z [INFO] Pulling docker image: ***/testcontainers/ryuk:0.11.0. Please be patient; this may take some time but only needs to be done once.
2025-05-14T06:22:06.5352835Z [INFO] Starting to pull image
2025-05-14T06:22:06.5530327Z [INFO] Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
2025-05-14T06:22:06.8369589Z [INFO] Pulling image layers:  2 pending,  1 downloaded,  0 extracted, (142 KB/? MB)
2025-05-14T06:22:06.8785032Z [INFO] Pulling image layers:  1 pending,  2 downloaded,  0 extracted, (142 KB/? MB)
2025-05-14T06:22:06.9158974Z [INFO] Pulling image layers:  0 pending,  3 downloaded,  0 extracted, (142 KB/10 MB)
2025-05-14T06:22:07.1016868Z [INFO] Pulling image layers:  0 pending,  3 downloaded,  1 extracted, (3 MB/10 MB)
2025-05-14T06:22:07.7733642Z [INFO] Pulling image layers:  0 pending,  3 downloaded,  2 extracted, (4 MB/10 MB)
2025-05-14T06:22:07.8814497Z [INFO] Pulling image layers:  0 pending,  3 downloaded,  3 extracted, (10 MB/10 MB)
2025-05-14T06:22:07.8879526Z [INFO] Pull complete. 3 layers, pulled in 1s (downloaded 10 MB at 10 MB/s)
2025-05-14T06:22:07.8880792Z [INFO] Image ***/testcontainers/ryuk:0.11.0 pull took PT1.663364457S
2025-05-14T06:22:07.9109036Z [INFO] Creating container for image: ***/testcontainers/ryuk:0.11.0
2025-05-14T06:22:07.9814544Z [INFO] Container ***/testcontainers/ryuk:0.11.0 is starting: b01c3070aa93322c53332f098f2b1b746e1f37f88c1b5beb87de6db2bf8aab30
2025-05-14T06:22:09.4964047Z [INFO] Container ***/testcontainers/ryuk:0.11.0 started in PT1.585396658S
2025-05-14T06:22:09.5020261Z [INFO] Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
2025-05-14T06:22:09.5021898Z [INFO] Checking the system...
2025-05-14T06:22:09.5911890Z [INFO] ✔︎ Docker server version should be at least 1.6.0
2025-05-14T06:22:09.5912919Z [INFO] Pulling docker image: ***/postgres:14.12. Please be patient; this may take some time but only needs to be done once.
2025-05-14T06:22:09.6794284Z [INFO] Starting to pull image
2025-05-14T06:22:09.6800857Z [INFO] Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
```

Second time in the same build.

```
2025-05-14T06:39:37.8546387Z [ERROR] Failed to execute goal org.codehaus.gmavenplus:gmavenplus-plugin:4.2.0:execute (postgres-container-start) on project aspect-materials-composition-repository: Error occurred while calling a method on a Groovy class from classpath. InvocationTargetException: Container startup failed for image ***/postgres:14.12: Retry limit hit with exception: Could not create/start container: Status 404: {"message":"No such image: ***/postgres:14.12"} -> [Help 1]
2025-05-14T06:39:37.8548116Z org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.gmavenplus:gmavenplus-plugin:4.2.0:execute (postgres-container-start) on project aspect-materials-composition-repository: Error occurred while calling a method on a Groovy class from classpath.
2025-05-14T06:39:37.8549321Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
2025-05-14T06:39:37.8553427Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
2025-05-14T06:39:37.8554481Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
2025-05-14T06:39:37.8555350Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
2025-05-14T06:39:37.8556247Z     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
2025-05-14T06:39:37.8557163Z     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
2025-05-14T06:39:37.8558279Z     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
2025-05-14T06:39:37.8559199Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
2025-05-14T06:39:37.8560169Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
2025-05-14T06:39:37.8561168Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
2025-05-14T06:39:37.8562258Z     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
2025-05-14T06:39:37.8563225Z     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
2025-05-14T06:39:37.8564125Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
2025-05-14T06:39:37.8564924Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
2025-05-14T06:39:37.8565701Z     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
2025-05-14T06:39:37.8566534Z     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
2025-05-14T06:39:37.8567813Z     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
2025-05-14T06:39:37.8568733Z     at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
2025-05-14T06:39:37.8569557Z     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
2025-05-14T06:39:37.8570348Z     at java.lang.reflect.Method.invoke (Method.java:580)
2025-05-14T06:39:37.8571067Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
2025-05-14T06:39:37.8571834Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
2025-05-14T06:39:37.8572904Z     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
2025-05-14T06:39:37.8573684Z     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
2025-05-14T06:39:37.8574449Z Caused by: org.apache.maven.plugin.MojoExecutionException: Error occurred while calling a method on a Groovy class from classpath.
2025-05-14T06:39:37.8575218Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.doExecute (ExecuteMojo.java:183)
2025-05-14T06:39:37.8575958Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.execute (ExecuteMojo.java:120)
2025-05-14T06:39:37.8576737Z     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
2025-05-14T06:39:37.8577534Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
2025-05-14T06:39:37.8578485Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
2025-05-14T06:39:37.8579264Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
2025-05-14T06:39:37.8580021Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
2025-05-14T06:39:37.8580874Z     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
2025-05-14T06:39:37.8581710Z     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
2025-05-14T06:39:37.8582659Z     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
2025-05-14T06:39:37.8583601Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
2025-05-14T06:39:37.8584535Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
2025-05-14T06:39:37.8585473Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
2025-05-14T06:39:37.8586456Z     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
2025-05-14T06:39:37.8587397Z     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
2025-05-14T06:39:37.8588425Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
2025-05-14T06:39:37.8589290Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
2025-05-14T06:39:37.8590054Z     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
2025-05-14T06:39:37.8590783Z     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
2025-05-14T06:39:37.8591531Z     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
2025-05-14T06:39:37.8592278Z     at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
2025-05-14T06:39:37.8593036Z     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
2025-05-14T06:39:37.8593841Z     at java.lang.reflect.Method.invoke (Method.java:580)
2025-05-14T06:39:37.8594665Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
2025-05-14T06:39:37.8595575Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
2025-05-14T06:39:37.8596447Z     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
2025-05-14T06:39:37.8597334Z     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
2025-05-14T06:39:37.8598354Z Caused by: java.lang.reflect.InvocationTargetException
2025-05-14T06:39:37.8599111Z     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:118)
2025-05-14T06:39:37.8599869Z     at java.lang.reflect.Method.invoke (Method.java:580)
2025-05-14T06:39:37.8600602Z     at org.codehaus.gmavenplus.util.ReflectionUtils.invokeMethod (ReflectionUtils.java:216)
2025-05-14T06:39:37.8601381Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.executeScripts (ExecuteMojo.java:259)
2025-05-14T06:39:37.8602433Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.doExecute (ExecuteMojo.java:179)
2025-05-14T06:39:37.8603405Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.execute (ExecuteMojo.java:120)
2025-05-14T06:39:37.8604307Z     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
2025-05-14T06:39:37.8605247Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
2025-05-14T06:39:37.8606127Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
2025-05-14T06:39:37.8607040Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
2025-05-14T06:39:37.8608090Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
2025-05-14T06:39:37.8608939Z     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
2025-05-14T06:39:37.8609873Z     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
2025-05-14T06:39:37.8610829Z     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
2025-05-14T06:39:37.8611801Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
2025-05-14T06:39:37.8612769Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
2025-05-14T06:39:37.8613739Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
2025-05-14T06:39:37.8614716Z     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
2025-05-14T06:39:37.8615741Z     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
2025-05-14T06:39:37.8616660Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
2025-05-14T06:39:37.8617527Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
2025-05-14T06:39:37.8618539Z     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
2025-05-14T06:39:37.8619278Z     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
2025-05-14T06:39:37.8619992Z     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
2025-05-14T06:39:37.8620744Z     at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
2025-05-14T06:39:37.8621682Z     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
2025-05-14T06:39:37.8622551Z     at java.lang.reflect.Method.invoke (Method.java:580)
2025-05-14T06:39:37.8623395Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
2025-05-14T06:39:37.8624319Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
2025-05-14T06:39:37.8625169Z     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
2025-05-14T06:39:37.8626118Z     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
2025-05-14T06:39:37.8627100Z Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image ***/postgres:14.12
2025-05-14T06:39:37.8628153Z     at org.testcontainers.containers.GenericContainer.doStart (GenericContainer.java:351)
2025-05-14T06:39:37.8628998Z     at org.testcontainers.containers.GenericContainer.start (GenericContainer.java:322)
2025-05-14T06:39:37.8629856Z     at org.codehaus.groovy.vmplugin.v8.IndyInterface.fromCache (IndyInterface.java:321)
2025-05-14T06:39:37.8630633Z     at startPostgres.run (startPostgres.groovy:14)
2025-05-14T06:39:37.8631353Z     at groovy.lang.GroovyShell.evaluate (GroovyShell.java:460)
2025-05-14T06:39:37.8632120Z     at groovy.lang.GroovyShell.evaluate (GroovyShell.java:504)
2025-05-14T06:39:37.8633018Z     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
2025-05-14T06:39:37.8633899Z     at java.lang.reflect.Method.invoke (Method.java:580)
2025-05-14T06:39:37.8635074Z     at org.codehaus.gmavenplus.util.ReflectionUtils.invokeMethod (ReflectionUtils.java:216)
2025-05-14T06:39:37.8635985Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.executeScripts (ExecuteMojo.java:259)
2025-05-14T06:39:37.8636840Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.doExecute (ExecuteMojo.java:179)
2025-05-14T06:39:37.8637688Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.execute (ExecuteMojo.java:120)
2025-05-14T06:39:37.8638800Z     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
2025-05-14T06:39:37.8639802Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
2025-05-14T06:39:37.8640680Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
2025-05-14T06:39:37.8641574Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
2025-05-14T06:39:37.8642497Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
2025-05-14T06:39:37.8643483Z     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
2025-05-14T06:39:37.8644378Z     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
2025-05-14T06:39:37.8645231Z     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
2025-05-14T06:39:37.8646181Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
2025-05-14T06:39:37.8647158Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
2025-05-14T06:39:37.8648336Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
2025-05-14T06:39:37.8649328Z     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
2025-05-14T06:39:37.8650335Z     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
2025-05-14T06:39:37.8651235Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
2025-05-14T06:39:37.8652093Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
2025-05-14T06:39:37.8652899Z     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
2025-05-14T06:39:37.8653636Z     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
2025-05-14T06:39:37.8654404Z     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
2025-05-14T06:39:37.8655085Z     at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
2025-05-14T06:39:37.8655931Z     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
2025-05-14T06:39:37.8656781Z     at java.lang.reflect.Method.invoke (Method.java:580)
2025-05-14T06:39:37.8657637Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
2025-05-14T06:39:37.8658748Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
2025-05-14T06:39:37.8659728Z     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
2025-05-14T06:39:37.8660523Z     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
2025-05-14T06:39:37.8661319Z Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
2025-05-14T06:39:37.8662164Z     at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess (Unreliables.java:88)
2025-05-14T06:39:37.8663038Z     at org.testcontainers.containers.GenericContainer.doStart (GenericContainer.java:336)
2025-05-14T06:39:37.8663873Z     at org.testcontainers.containers.GenericContainer.start (GenericContainer.java:322)
2025-05-14T06:39:37.8664775Z     at org.codehaus.groovy.vmplugin.v8.IndyInterface.fromCache (IndyInterface.java:321)
2025-05-14T06:39:37.8665620Z     at startPostgres.run (startPostgres.groovy:14)
2025-05-14T06:39:37.8666666Z     at groovy.lang.GroovyShell.evaluate (GroovyShell.java:460)
2025-05-14T06:39:37.8667575Z     at groovy.lang.GroovyShell.evaluate (GroovyShell.java:504)
2025-05-14T06:39:37.8668954Z     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
2025-05-14T06:39:37.8669839Z     at java.lang.reflect.Method.invoke (Method.java:580)
2025-05-14T06:39:37.8670700Z     at org.codehaus.gmavenplus.util.ReflectionUtils.invokeMethod (ReflectionUtils.java:216)
2025-05-14T06:39:37.8671535Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.executeScripts (ExecuteMojo.java:259)
2025-05-14T06:39:37.8672295Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.doExecute (ExecuteMojo.java:179)
2025-05-14T06:39:37.8673067Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.execute (ExecuteMojo.java:120)
2025-05-14T06:39:37.8673866Z     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
2025-05-14T06:39:37.8674692Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
2025-05-14T06:39:37.8675624Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
2025-05-14T06:39:37.8676566Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
2025-05-14T06:39:37.8677520Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
2025-05-14T06:39:37.8678547Z     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
2025-05-14T06:39:37.8679418Z     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
2025-05-14T06:39:37.8680424Z     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
2025-05-14T06:39:37.8681438Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
2025-05-14T06:39:37.8682451Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
2025-05-14T06:39:37.8683521Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
2025-05-14T06:39:37.8684548Z     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
2025-05-14T06:39:37.8685419Z     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
2025-05-14T06:39:37.8686252Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
2025-05-14T06:39:37.8687121Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
2025-05-14T06:39:37.8688088Z     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
2025-05-14T06:39:37.8688940Z     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
2025-05-14T06:39:37.8689743Z     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
2025-05-14T06:39:37.8690544Z     at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
2025-05-14T06:39:37.8691374Z     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
2025-05-14T06:39:37.8692214Z     at java.lang.reflect.Method.invoke (Method.java:580)
2025-05-14T06:39:37.8693079Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
2025-05-14T06:39:37.8694002Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
2025-05-14T06:39:37.8694848Z     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
2025-05-14T06:39:37.8695722Z     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
2025-05-14T06:39:37.8696508Z Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
2025-05-14T06:39:37.8697451Z     at org.testcontainers.containers.GenericContainer.tryStart (GenericContainer.java:556)
2025-05-14T06:39:37.8698593Z     at org.testcontainers.containers.GenericContainer.lambda$doStart$0 (GenericContainer.java:346)
2025-05-14T06:39:37.8699881Z     at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess (Unreliables.java:81)
2025-05-14T06:39:37.8700780Z     at org.testcontainers.containers.GenericContainer.doStart (GenericContainer.java:336)
2025-05-14T06:39:37.8701587Z     at org.testcontainers.containers.GenericContainer.start (GenericContainer.java:322)
2025-05-14T06:39:37.8702373Z     at org.codehaus.groovy.vmplugin.v8.IndyInterface.fromCache (IndyInterface.java:321)
2025-05-14T06:39:37.8703075Z     at startPostgres.run (startPostgres.groovy:14)
2025-05-14T06:39:37.8703755Z     at groovy.lang.GroovyShell.evaluate (GroovyShell.java:460)
2025-05-14T06:39:37.8704432Z     at groovy.lang.GroovyShell.evaluate (GroovyShell.java:504)
2025-05-14T06:39:37.8705189Z     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
2025-05-14T06:39:37.8705931Z     at java.lang.reflect.Method.invoke (Method.java:580)
2025-05-14T06:39:37.8706659Z     at org.codehaus.gmavenplus.util.ReflectionUtils.invokeMethod (ReflectionUtils.java:216)
2025-05-14T06:39:37.8707457Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.executeScripts (ExecuteMojo.java:259)
2025-05-14T06:39:37.8708372Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.doExecute (ExecuteMojo.java:179)
2025-05-14T06:39:37.8709109Z     at org.codehaus.gmavenplus.mojo.ExecuteMojo.execute (ExecuteMojo.java:120)
2025-05-14T06:39:37.8709916Z     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
2025-05-14T06:39:37.8710739Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
2025-05-14T06:39:37.8711563Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
2025-05-14T06:39:37.8712355Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
2025-05-14T06:39:37.8713139Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
2025-05-14T06:39:37.8713930Z     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
2025-05-14T06:39:37.8714696Z     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
2025-05-14T06:39:37.8715505Z     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
2025-05-14T06:39:37.8716397Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
2025-05-14T06:39:37.8717359Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
2025-05-14T06:39:37.8718548Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
2025-05-14T06:39:37.8719413Z     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
2025-05-14T06:39:37.8720418Z     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
2025-05-14T06:39:37.8721350Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
2025-05-14T06:39:37.8722204Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
2025-05-14T06:39:37.8722983Z     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
2025-05-14T06:39:37.8723799Z     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
2025-05-14T06:39:37.8724655Z     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
2025-05-14T06:39:37.8725506Z     at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
2025-05-14T06:39:37.8726438Z     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
2025-05-14T06:39:37.8727176Z     at java.lang.reflect.Method.invoke (Method.java:580)
2025-05-14T06:39:37.8728119Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
2025-05-14T06:39:37.8728956Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
2025-05-14T06:39:37.8730163Z     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
2025-05-14T06:39:37.8731098Z     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
2025-05-14T06:39:37.8732100Z Caused by: com.github.dockerjava.api.exception.NotFoundException: Status 404: {"message":"No such image: ***/postgres:14.12"}
```

Anyone have any idea how this could happen? And why it only happens sometimes?
